### ISSUE TITLE
Consider normalizing path separators in setenv for PATH variables

### ISSUE THREAD

**Issue by pytoxbot** [tox-dev/tox#344]
- Bitbucket: https://bitbucket.org/hpk42/tox/issue/344
- Originally reported by: [rogalski](http://bitbucket.org/rogalski)
- Originally created at: 2016-07-12T19:15:23.358

Consider following extract from tox.ini:

```
#!ini

[testenv]
setenv =
         PYTHONPATH={toxinidir}/a/;{toxinidir}/b/

```

Path separator used for PYTHONPATH (and PATH variables) is platform specific - so invocation above would work on Windows but would not on POSIX system. While usually it's solved via testing matrix, I'm not sure if that's a correct approach here. User intention is to add multiple paths to environment variable, and copy-pasting same path list and changing only their separator is a clear violation of DRY.

I was thinking about few strategies:
- always normalizing separators in PATH and PYTHONPATH variables (from colon or semicolon to `os.pathsep`). I'm not sure if it's smart to maintain any kind of whitelist in docs. Also, explicit is better than implicit, and here conversion would be performed implicitly.
- export `{os.pathsep}` as possible substitute - explicit, easy, but results in kind of ugly looking tox.ini
- add optional field in tox.ini `env_path_vars = PATH,PYTHONPATH` - normalization would be performed only for explicitly selected environment variables. It looks clean, but adds additional config parameter.

I'd be happy to implement any of these (if proposal will be accepted) - the question is, which proposal do you like the most?

Regards,
Lukasz

**Comment by rogalski** [comment]
What's the follow-up on this one? Is a proposal rejected? See a pull request for potential implementation: https://bitbucket.org/hpk42/tox/pull-requests/203/normalize-path-separators-for-setenv/diff

**Comment by obestwalter** [comment]
Hi @rogalski whoops - did not see that someone was waiting for my input here - sorry. I don't really have an opinion on this one yet, but the PR you originally opened on bitbucket is now obsolete, so if you could reopen it against the moved repo we would have a base for the conversation. My initial feeling would also be with using ":" as separator and have some way of signalling that it should be normalized to be portable.

**Comment by rogalski** [comment]
It may be closed since #402 is merged.

**Comment by obestwalter** [comment]
@rogalski Thanks!
